sdk/keyvault/azsecrets/client.go

```diff
@@ -330,93 +379,107 @@ type GetDeletedSecretResponse struct {
+	// placeholder for future optional parameters
 }
 
 func (u UpdateSecretPropertiesOptions) toGenerated() *internal.KeyVaultClientUpdateSecretOptions {
 	return &internal.KeyVaultClientUpdateSecretOptions{}
 }
 
-func (u UpdateSecretPropertiesOptions) toGeneratedProperties() internal.SecretUpdateParameters {
-	return internal.SecretUpdateParameters{
-		ContentType:      u.ContentType,
-		SecretAttributes: u.Properties.toGenerated(),
-		Tags:             convertToGeneratedMap(u.Tags),
-	}
-}
-
 // UpdateSecretPropertiesResponse contains the underlying response object for the UpdateSecretProperties method
 type UpdateSecretPropertiesResponse struct {
 	Secret
 }
 
 func updateSecretPropertiesResponseFromGenerated(i internal.KeyVaultClientUpdateSecretResponse) UpdateSecretPropertiesResponse {
+	vaultURL, name, version := shared.ParseID(i.ID)
 	return UpdateSecretPropertiesResponse{
 		Secret: Secret{
-			Properties:  secretPropertiesFromGenerated(i.Attributes),
-			ContentType: i.ContentType,
-			ID:          i.ID,
-			Tags:        convertPtrMap(i.Tags),
-			Value:       i.Value,
-			KeyID:       i.Kid,
-			Managed:     i.Managed,
+			Properties: &Properties{
+				ContentType:     i.ContentType,
+				CreatedOn:       i.Attributes.Created,
+				Enabled:         i.Attributes.Enabled,
+				ExpiresOn:       i.Attributes.Expires,
+				IsManaged:       i.Managed,
+				KeyID:           i.Kid,
+				NotBefore:       i.Attributes.NotBefore,
+				RecoverableDays: i.Attributes.RecoverableDays,
+				RecoveryLevel:   (*string)(i.Attributes.RecoveryLevel),
+				Tags:            convertPtrMap(i.Tags),
+				UpdatedOn:       i.Attributes.Updated,
+				VaultURL:        vaultURL,
+				Version:         version,
+				Name:            name,
+			},
+			ID:    i.ID,
+			Name:  name,
+			Value: i.Value,
 		},
 	}
 }
 
 // UpdateSecretProperties updates the attributes associated with a specified secret in a given key vault. The update
 // operation changes specified attributes of an existing stored secret, attributes that are not specified in the
 // request are left unchanged. The value of a secret itself cannot be changed. This operation requires the secrets/set permission.
-func (c *Client) UpdateSecretProperties(ctx context.Context, secretName string, options *UpdateSecretPropertiesOptions) (UpdateSecretPropertiesResponse, error) {
-	if options == nil {
-		options = &UpdateSecretPropertiesOptions{}
+func (c *Client) UpdateSecretProperties(ctx context.Context, secret Secret, options *UpdateSecretPropertiesOptions) (UpdateSecretPropertiesResponse, error) {
+	name, version := "", ""
+	if secret.Properties != nil && secret.Properties.Name != nil {
+		name = *secret.Properties.Name
+	}
+	if secret.Properties != nil && secret.Properties.Version != nil {
+		version = *secret.Properties.Version
 	}
 
 	resp, err := c.kvClient.UpdateSecret(
 		ctx,
 		c.vaultUrl,
-		secretName,
-		options.Version,
-		options.toGeneratedProperties(),
-		options.toGenerated(),
+		name,
```

Comment:
If a caller constructed the `Secret` instead of getting it from another method, `name` could be empty, in which case the request can't succeed. I guess the vault would respond 404, maybe with a baffling HTML payload; the caller could ultimately get a serialization error. Should we forego the request and return an error when `name == ""`?

Reply:
There is a check in the generated code that validates name is not empty. It returns `errors.New("parameter secretName cannot be empty")`

Reply:
That's good, if awkward because there's no public `secretName`. I wouldn't block the release on the error message in this unlikely scenario though